infoschema/tables.go

```diff
@@ -261,19 +261,38 @@ func buildColumnInfo(col columnInfo) *model.ColumnInfo {
 
 func buildTableMeta(tableName string, cs []columnInfo) *model.TableInfo {
 	cols := make([]*model.ColumnInfo, 0, len(cs))
+	primaryIndices := make([]*model.IndexInfo, 0, 1)
+	tblInfo := &model.TableInfo{
+		Name:    model.NewCIStr(tableName),
+		State:   model.StatePublic,
+		Charset: mysql.DefaultCharset,
+		Collate: mysql.DefaultCollationName,
+	}
 	for _, c := range cs {
+		if c.flag&mysql.PriKeyFlag == 1 {
+			if c.tp == mysql.TypeInt24 {
+				tblInfo.PKIsHandle = true
+			} else {
+				tblInfo.PKIsHandle = true
+				index := &model.IndexInfo{
+					Name:    model.NewCIStr("primary"),
+					State:   model.StatePublic,
+					Primary: true,
+					Unique:  true,
+					Columns: []*model.IndexColumn{
+						{Name: model.NewCIStr(c.name), Offset: 0}},
```

Comment:
```sql
{
							Name:   model.NewCIStr(c.name),
							Offset: offset,
							Length: types.UnspecifiedLength,

},
```

The offset is not 0, it should be the offset of the columns. use `for offset, c := range cs` in lin 271.

Reply:
and the `Length` in this case, should be `UnspecifiedLength`.